test/setup/main.go

```diff
@@ -0,0 +1,230 @@
+	var clusterMonitoringConfigYAML bytes.Buffer
+	if err := clusterMonitoringConfigTemplate.Execute(&clusterMonitoringConfigYAML, o); err != nil {
+		return err
+	}
+	var userWorkloadMonitoringConfigYAML bytes.Buffer
+	if err := userWorkloadMonitoringConfigTemplate.Execute(&userWorkloadMonitoringConfigYAML, o); err != nil {
+		return err
+	}
+
+	// Collect remote write config if specified
+	var username, password string
+	if len(o.RemoteWriteURL) > 0 {
+		log.Info("remote write will be enabled")
+		username = o.RemoteWriteUsername
+		if len(o.RemoteWriteUsernameFile) > 0 {
+			u, err := ioutil.ReadFile(o.RemoteWriteUsernameFile)
+			if err != nil {
+				return err
+			}
+			username = string(u)
+		}
+		password = o.RemoteWritePassword
+		if len(o.RemoteWritePasswordFile) > 0 {
+			p, err := ioutil.ReadFile(o.RemoteWritePasswordFile)
+			if err != nil {
+				return err
+			}
+			password = string(p)
+		}
+		if len(username) == 0 {
+			return fmt.Errorf("username is required")
+		}
+		if len(password) == 0 {
+			return fmt.Errorf("password is required")
+		}
+	}
+
+	// Install the remote write secret referenced by the remote write configuration
+	clusterMonitoringRemoteWriteSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "openshift-monitoring",
+			Name:      "remote-write-creds",
+		},
+	}
+	if result, err := controllerutil.CreateOrUpdate(ctx, k, clusterMonitoringRemoteWriteSecret, func() error {
+		clusterMonitoringRemoteWriteSecret.Data = map[string][]byte{
+			"username": []byte(username),
+			"password": []byte(password),
+		}
+		return nil
+	}); err != nil {
+		return err
+	} else {
+		log.Info("updated cluster monitoring remote write secret", "result", result)
+	}
+
+	// Enable user workload monitoring and remote write from the cluster monitoring
+	// stack
+	clusterMonitoringConfig := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "openshift-monitoring",
+			Name:      "cluster-monitoring-config",
+		},
+	}
+	if result, err := controllerutil.CreateOrUpdate(ctx, k, clusterMonitoringConfig, func() error {
+		clusterMonitoringConfig.Data = map[string]string{
+			"config.yaml": clusterMonitoringConfigYAML.String(),
+		}
+		return nil
+	}); err != nil {
+		return err
+	} else {
+		log.Info("updated cluster monitoring config", "result", result)
+	}
+
+	// Wait for the user workload namespace to exist, which is a loose indicator
+	// that user workload monitoring was enabled
+	userWorkloadMonitoringNamespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "openshift-user-workload-monitoring",
+		},
+	}
+	err := wait.PollUntil(1*time.Second, func() (done bool, err error) {
+		err = k.Get(ctx, client.ObjectKeyFromObject(userWorkloadMonitoringNamespace), userWorkloadMonitoringNamespace)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Info("waiting for user workload monitoring namespace to exist", "namespace", userWorkloadMonitoringNamespace.Name)
```

Comment:
Maybe call it by its name (`openshift-user-workload-monitoring`) rather than a description?